federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProvider.java

```diff
@@ -168,6 +168,10 @@ public LDAPStorageUserManager getUserManager() {
 
     @Override
     public UserModel validate(RealmModel realm, UserModel local) {
+        if(!this.model.isValidateOnAccess()) {
+            return local;
+        }
```

Comment:
Returning `local` and not returning a proxy for the LDAP, thereby any write access to the entity is not propagated to LDAP. This violates IMHO the API of this method.

Another possibility would be to wrap the local in `ReadOnlyUserModelDelegate`, which would avoid loosing any changes, but would still possibly violate some caller's expectations.

The one thing that would probably match a caller's expectation reasonably well would be a lazy-loading of the LDAP element on the first write access.

Reply:
Thank you @ahus1 for your comment and support on this. Now there is a proxy-model that allowes writes on demand.